src/sage/modular/hypergeometric_motive.py

```diff
@@ -1753,11 +1793,23 @@ def euler_factor(self, t, p, cache_p=False):
             raise ValueError('p not prime')
         if not all(x.denominator() % p for x in self._alpha + self._beta):
             raise NotImplementedError('p is wild')
-        if (t.numerator() * t.denominator() % p == 0 or (t - 1) % p == 0):
+        e = t.valuation(p)
+        if (t.numerator()*t.denominator() % p == 0):
```

Comment:
```suggestion
        if t.numerator()*t.denominator() % p == 0:
```
Not needed in Python.